packages/big-design/src/components/FeatureTag/FeatureTag.tsx

```diff
@@ -0,0 +1,32 @@
+import { IconProps } from '@bigcommerce/big-design-icons';
+import React, { AnchorHTMLAttributes, FunctionComponent, ReactElement } from 'react';
+
+import { StyledFeatureTag, StyledFeatureTagIcon, StyledFeatureTagLabel } from './styled';
+
+export interface FeatureTagProps extends AnchorHTMLAttributes<HTMLAnchorElement> {
+  icon?: ReactElement<IconProps>;
+  isActive?: boolean;
+  label: string;
+}
+
+export const FeatureTag: FunctionComponent<FeatureTagProps> = ({
+  label,
+  icon,
+  isActive,
+  href,
+  target,
+}) => {
+  return label ? (
+    <StyledFeatureTag
+      className={isActive ? 'active' : ''}
+      href={href}
+      target={target}
+      title={label}
+    >
+      {icon ? <StyledFeatureTagIcon>{icon}</StyledFeatureTagIcon> : null}
+      <StyledFeatureTagLabel>{label}</StyledFeatureTagLabel>
+    </StyledFeatureTag>
+  ) : null;
+};
+
+FeatureTag.displayName = 'FeatureTag';
```

Comment:
Ton's of things here:
1. For icon props, we just use `ReactNode` as the type. Allows consumers to pass in high-order components/icons if needed.
2. I know this might be a contention point, but let's remove the `isActive` prop. It feels like it's a link state, rather than prop state here.
3. Use `ComponentPropsWithoutRef` instead of `HTML<Element>Attributes` to extend from.
4. Avoid using `FunctionComponent`, instead use `({ some, prop, ...props }: Props) => {...}` as React now implies it's a functional component.
5. You'll see in `styled.tsx` some simplification there, which simplifies the HTML and styles in the component.
6. Spread the rest of the props as we are extending a link component.

```suggestion
import React, { ComponentPropsWithoutRef, forwardRef, ReactNode } from 'react';

import { StyledFeatureTag, StyledFeatureTagLabel } from './styled';

export interface FeatureTagProps extends ComponentPropsWithoutRef<'a'> {
  icon?: ReactNode;
  label: string;
}

export const FeatureTag = forwardRef<HTMLAnchorElement, FeatureTagProps>(
  ({ label, icon, ...props }, ref) => {
    return label ? (
      <StyledFeatureTag ref={ref} {...props}>
        {icon}
        <StyledFeatureTagLabel>{label}</StyledFeatureTagLabel>
      </StyledFeatureTag>
    ) : null;
  },
);

FeatureTag.displayName = 'FeatureTag';
```

Reply:
Hi @chanceaclark .

Good pointers, but regarding the isActive prop, it denotes that it is the current category that you are browsing, just like a nav menu would highlight the current page. Would it be better to change the prop name to `isCurrent`?

Reply:
Can you provide an example on how app developers are going to use this? To me it seems like `FeatureTag`s list the capabilities of the app without linking anywhere. It's not clear to me how an app developer should think about using this component, specifically linking to what/where?

If we are trying to build a component that has active click states, then we need to redesign the concept of this component to act more like a tabable navigation instead as there are other accessibility roles we should handle as a library.

Reply:
Sure, the iadea now is the following. When seeing a particular integration within the control panel, there will be feature tags representing the type and category of this particular app or integration. This could be things like, headless, payments, checkout and so on.

If you were consuming that particular content within the context of say "headless", then the "headless" feature tag would be marked as "current" or "active", kind of like filter chips on the sidebar of our storefronts when a filtering occurs.

The feature tag has always been thought as a navigational item, that's why the current/active state was set in place.

Reply:
Following Figma asset as source of truth:
https://www.figma.com/design/ig91vdN7fcn1sCPgj68jA0/BigDesign-Component-Library?node-id=1446-29&node-type=canvas&t=ExzdQvs66DfnrxnX-0

Reply:
After discussion with both @BC-Turner and @chanceaclark , the direction of this component is set to be a tag and not a link.